src/plugins/application_config/README.md

```diff
@@ -0,0 +1,112 @@
+# ApplicationConfig Plugin
+
+An OpenSearch Dashboards plugin for application configuration service and OpenSearch implementation.
+
+---
+
+## Introduction
+
+This plugin introduces the support of dynamic application configurations as opposed to the existing static configuration in OSD YAML file `opensearch_dashboards.yml`. It stores the configuration in an index whose default name is `.opensearch_dashboards_config` and could be customized through the key `opensearchDashboards.configIndex` in OSD YAML file. Initially the new index does not exist. Only OSD users who need dynamic configurations will create it.
+
+It also provides an interface `ConfigurationClient` for future extensions of external configuration clients. A default implementation based on OpenSearch as database is used.
+
+This plugin is disabled by default.
+
+## Configuration
+
+OSD users who want to set up application configurations will first need to enable this plugin by the following line in OSD YML.
+
+```
+application_config.enabled: true
```

Comment:
I didn't see entry for this configuration in YML file in this PR. Is it covered in another PR?

Reply:
Added.